src/System.Collections.Concurrent/tests/BlockingCollectionTests.cs

```diff
@@ -48,5 +48,1967 @@ public static void TestBlockingCollectionBasic()
+
+        /// <summary>Validates the Dispose() method.</summary>
+        /// <returns>True if test succeeded, false otherwise.</returns>
+        [Fact]
+        [OuterLoop]
+        public static void RunBlockingCollectionTest4_Dispose()
+        {
+            BlockingCollection<int> blockingCollection = ConstructBlockingCollection<int>();
+
+            blockingCollection.Dispose();
+            bool testSuceeded = false;
+            int numOfExceptionsThrown = 0;
+            int numOfTests = 0;
+
+            try
+            {
+                numOfTests++;
+                blockingCollection.Add(default(int));
+            }
+            catch (ObjectDisposedException)
+            {
+                numOfExceptionsThrown++;
+            }
+
+            try
+            {
+                numOfTests++;
+                blockingCollection.TryAdd(default(int));
+            }
+            catch (ObjectDisposedException)
+            {
+                numOfExceptionsThrown++;
+            }
+
+            try
+            {
+                numOfTests++;
+                blockingCollection.TryAdd(default(int), 1);
+            }
+            catch (ObjectDisposedException)
+            {
+                numOfExceptionsThrown++;
+            }
+
+            try
+            {
+                numOfTests++;
+                blockingCollection.TryAdd(default(int), new TimeSpan(1));
+            }
+            catch (ObjectDisposedException)
+            {
+                numOfExceptionsThrown++;
+            }
+
+            int item;
+            try
+            {
+                numOfTests++;
+                blockingCollection.Take();
+            }
+            catch (ObjectDisposedException)
+            {
+                numOfExceptionsThrown++;
+            }
+
+            try
+            {
+                numOfTests++;
+                blockingCollection.TryTake(out item);
+            }
+            catch (ObjectDisposedException)
+            {
+                numOfExceptionsThrown++;
+            }
+
+            try
+            {
+                numOfTests++;
+                blockingCollection.TryTake(out item, 1);
+            }
+            catch (ObjectDisposedException)
+            {
+                numOfExceptionsThrown++;
+            }
+
+            try
+            {
+                numOfTests++;
+                blockingCollection.TryTake(out item, new TimeSpan(1));
+            }
+            catch (ObjectDisposedException)
+            {
+                numOfExceptionsThrown++;
+            }
+
+            const int NUM_OF_COLLECTIONS = 10;
+            BlockingCollection<int>[] blockingCollections = new BlockingCollection<int>[NUM_OF_COLLECTIONS];
+            for (int i = 0; i < NUM_OF_COLLECTIONS - 1; ++i)
+            {
+                blockingCollections[i] = ConstructBlockingCollection<int>(-1);
+            }
+
+            blockingCollections[NUM_OF_COLLECTIONS - 1] = blockingCollection;
```

Comment:
This whole section of code is repeatedly executing the same boilerplate, just with different operations each time.  It could be simplified with either an array of delegates, each representing a particular operation, that we then iterate through, or with a helper function that accepts a delegate and provides the boilerplate and which we then call once for each operation.